apps/server/src/adapters/WebsocketAdapter.ts

```diff
@@ -256,6 +279,29 @@ export class SocketServer implements IAdapter {
     }
   }
 
+  public sendRuntimeStoreUpdate(keys: (keyof RuntimeStore)[], store: Partial<RuntimeStore>) {
```

Comment:
this logic looks a bit complicated. 
Can you please explain what it does (in code)?